transport/http2_server.go

```diff
@@ -893,11 +898,11 @@ func (t *http2Server) Write(s *Stream, data []byte, opts *Options) (err error) {
 		// Reset ping strikes when sending data since this might cause
 		// the peer to send ping.
 		atomic.StoreUint32(&t.resetPingStrikes, 1)
-		if err := t.framer.writeData(forceFlush, s.id, false, p); err != nil {
+		if err := t.framer.writeData(forceFlush && !opts.Delay, s.id, false, p); err != nil {
```

Comment:
Might be wrong, but for the scenario in which the client has an inbound flow control which is less than the size of the server's response message, IIC this change could cause deadlock, with the server running out of flow control and the client not having yet received anything.

This looks similar to the change in https://github.com/grpc/grpc-go/pull/973, which ran into that issue.